src/test/javascript/spec/service/submission.service.spec.ts

```diff
@@ -94,6 +102,92 @@ describe('Submission Service', () => {
+
+        const firstResult: Result = {
+            id: 3556,
+            resultString: '3 of 12 points',
+            score: 24,
+            rated: true,
+            hasFeedback: true,
+            hasComplaint: false,
+            feedbacks: [firstFeedback],
+        };
+
+        submission.results?.unshift(firstResult);
+
+        expect(secondFeedback.copiedFeedbackId).toBe(undefined);
+
+        const latestResultFeedbacks = getLatestSubmissionResult(submission)!.feedbacks!;
+        latestResultFeedbacks?.push(secondFeedback);
+
+        service.handleFeedbackCorrectionRoundTag(0, submission);
+        expect(secondFeedback.copiedFeedbackId).toBe(undefined);
+
+        service.handleFeedbackCorrectionRoundTag(1, submission);
+        expect(latestResultFeedbacks[0].copiedFeedbackId).toBe(undefined);
+        expect(secondFeedback.copiedFeedbackId).toBe(firstFeedback.id);
+
+        secondFeedback.text = 'Feedback changed';
+        service.handleFeedbackCorrectionRoundTag(2, submission);
+        expect(secondFeedback.copiedFeedbackId).toBe(undefined);
+    });
+
+    it('should convert results date from server', () => {
+        const dateStr = '2022-02-02T17:37:53.283Z';
+        const date = dayjs(dateStr);
+
+        const result = submission.results![0];
+        result.completionDate = dateStr as any; // String should be converted to proper type by the tested service.
+
+        service.convertResultsDateFromServer(submission.results);
+
+        expect(result.completionDate).toEqual(date);
+    });
+
+    it('should get submissions with complaints for tutor', fakeAsync(() => {
+        const exerciseId = 1;
+        const submissionDateStr = '2022-02-02T12:34:56.789Z';
+        const complaintSubmittedTimeStr = '2022-02-03T22:11:33.444Z';
+
+        const complaint: Complaint = {
+            submittedTime: complaintSubmittedTimeStr as any, // String should be converted to proper type by the tested service.
+        };
+        const returnedFromService: SubmissionWithComplaintDTO[] = [
+            {
+                submission,
+                complaint,
+            },
+        ];
+        submission.submissionDate = submissionDateStr as any; // String should be converted to proper type by the tested service.
+
+        service
+            .getSubmissionsWithComplaintsForTutor(exerciseId)
+            .pipe(take(1))
+            .subscribe((resp) => {
+                expect(resp.body).toHaveLength(1);
+                const submissionWithComplaint = resp.body![0];
+                expect(submissionWithComplaint.submission.submissionDate).toEqual(dayjs(submissionDateStr));
+                expect(submissionWithComplaint.complaint.submittedTime).toEqual(dayjs(complaintSubmittedTimeStr));
+            });
+        const req = httpMock.expectOne({ url: `api/exercises/${exerciseId}/submissions-with-complaints`, method: 'GET' });
+        req.flush(returnedFromService);
+        tick();
+    }));
+
     afterEach(() => {
```

Comment:
Can you move the `afterEach` to the top below the `beforeEach`? That way it would be more visible and would follow the usual structure of test files.

Reply:
Done.